### PR TITLE
KFSPTS-10032: POJO classes for Default KFS Account for AWS

### PR DESCRIPTION
**Pull request by ajd299**
- POJO Classes for DefaultKFSAccountForAWS
- Tests to Marshal/Unmarshal to/from class from string

### CODE REVIEW

**Comment by db483** [comment]
retest this please

**Comment by db483** [comment]
retest this please

**Comment by db483** [comment]
retest this please

**Comment by jhulslander** [comment]
retest this please

**Comment by db483** [comment]
retest this please

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
retest this please

File: src/test/java/edu/cornell/kfs/fp/batch/xml/DefaultKfsAccountForAwsMarshalTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    
+    @Test
+    public void verifyDefaultKfsAccountForAwsXmlCanBeMarshalled() throws JAXBException, IOException {
+        String marshalledXml = cUMarshalService.marshalObjectToXmlString(
+                DefaultKfsAccountForAwsWrapperFixture.generateExpectedDefaultKfsAccountForAwsResultWrapperObject());
+        marshalledXml = removeWhitespace(marshalledXml.substring(marshalledXml.indexOf('\n') + 1));
+
+        String expectedXmlExample = DefaultKfsAccountForAwsWrapperFixture.getXmlFromExampleFile();
+        expectedXmlExample = removeWhitespace(expectedXmlExample);
+        assertTrue("The Marshalled XML should be equal to the example XML", marshalledXml.equalsIgnoreCase(expectedXmlExample));
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/batch/xml/DefaultKfsAccountForAwsMarshalTest.java]: you could use StringUtils.equalsIgnoreCase to avoid NPEs here

File: src/test/java/edu/cornell/kfs/fp/batch/xml/DefaultKfsAccountForAwsMarshalTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    @Test
+    public void verifyDefaultKfsAccountForAwsXmlCanBeUnmarshalled() throws JAXBException, IOException {
+        DefaultKfsAccountForAwsResultWrapper unmarshalledObject =
+                cUMarshalService.unmarshalString(DefaultKfsAccountForAwsWrapperFixture.getXmlFromExampleFile(), DefaultKfsAccountForAwsResultWrapper.class);
+
+        assertTrue(isDefaultKfsAccountForAwsResultsWrapperEqual(unmarshalledObject,
+                DefaultKfsAccountForAwsWrapperFixture.generateExpectedDefaultKfsAccountForAwsResultWrapperObject()));
+    }
+
+    private boolean isDefaultKfsAccountForAwsResultsWrapperEqual(DefaultKfsAccountForAwsResultWrapper a, DefaultKfsAccountForAwsResultWrapper b) {
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/batch/xml/DefaultKfsAccountForAwsMarshalTest.java]: I would rename the input parameters to expectedWrapper and actualWrapper

File: src/test/java/edu/cornell/kfs/fp/batch/xml/DefaultKfsAccountForAwsMarshalTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        List<DefaultKfsAccountForAws> aList = a.getDefaultKfsAccountForAws();
+        List<DefaultKfsAccountForAws> bList = b.getDefaultKfsAccountForAws();
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/batch/xml/DefaultKfsAccountForAwsMarshalTest.java]: Similarly these two should be renamed to something more meaninful

**Comment by ajd299** [comment]
Thanks for the feedback, I will work on fixing these issues

File: src/test/java/edu/cornell/kfs/fp/batch/xml/fixture/DefaultKfsAccountForAwsWrapperFixture.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+public class DefaultKfsAccountForAwsWrapperFixture {
+
+    private static final String EXPECTED_XML_EXAMPLE_FILE_PATH = "src/test/resources/edu/cornell/kfs/fp/batch/xml/default-kfs-account-for-aws-test.xml";
+
+    public static DefaultKfsAccountForAwsResultWrapper generateExpectedDefaultKfsAccountForAwsResultWrapperObject() {
+        DefaultKfsAccountForAwsResultWrapper defaultKfsAccountForAwsResultWrapper = new DefaultKfsAccountForAwsResultWrapper();
+
+        ArrayList<DefaultKfsAccountForAws> defaultKfsAccountForAwsList = new ArrayList<>();
+        defaultKfsAccountForAwsList.add(buildDefaultKfsAccountForAws("999913976919", "1719999", "2016-04-26 18:43:03"));
```
[the remark this answers is not in the document]
Comment (ajd299) [reply src/test/java/edu/cornell/kfs/fp/batch/xml/fixture/DefaultKfsAccountForAwsWrapperFixture.java]: No, I sourced the examples from the kfsaws-support-api REST API Documentation in confluence.

**Comment by jhulslander** [comment]
one other thing, we should add a couple of unit tests to make sure the adapter can handle an empty date sting, and a poorly formatted date string.

**Comment by jhulslander** [comment]
retest this please

**Comment by ajd299** [comment]
I ended up creating AwsStringToJavaDateTimeAdapter to capture the time element. I also tried the  RiceDateTimeAdapter, but it didn't seem to handle blank/null values gracefully. Please code review again.

**Comment by ajd299** [comment]
retest this please

File: src/test/resources/edu/cornell/kfs/fp/batch/xml/default-kfs-account-for-aws-empty-date-test.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<result>
+    <row>
+        <aws_account>999913976919</aws_account>
+        <kfs_default_account>1719999</kfs_default_account>
+        <updated_at></updated_at>
```
Comment (jhulslander) [inline src/test/resources/edu/cornell/kfs/fp/batch/xml/default-kfs-account-for-aws-empty-date-test.xml]: can you add a row that doesn't have the updated_at tags at all?

File: src/test/java/edu/cornell/kfs/fp/batch/xml/fixture/DefaultKfsAccountForAwsWrapperFixture.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    private static final String BAD_DATE_XML_EXAMPLE_FILE_PATH = "src/test/resources/edu/cornell/kfs/fp/batch/xml/default-kfs-account-for-aws-bad-date-test.xml";
+    private static final String EMPTY_DATE_XML_EXAMPLE_FILE_PATH = "src/test/resources/edu/cornell/kfs/fp/batch/xml/default-kfs-account-for-aws-empty-date-test.xml";
+
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+
+    public static DefaultKfsAccountForAwsResultWrapper generateExpectedDefaultKfsAccountForAwsResultWrapper() {
+        DefaultKfsAccountForAwsResultWrapper defaultKfsAccountForAwsResultWrapper = new DefaultKfsAccountForAwsResultWrapper();
+
+        ArrayList<DefaultKfsAccountForAws> defaultKfsAccountForAwsList = new ArrayList<>();
+        defaultKfsAccountForAwsList.add(buildDefaultKfsAccountForAws("999913976919", "1719999", "2016-04-26 18:43:03"));
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/batch/xml/fixture/DefaultKfsAccountForAwsWrapperFixture.java]: Sorry for not mentioning the first time, but you could move all these values to private static final variables, and avoid duplicating these strings.

File: src/main/java/edu/cornell/kfs/sys/xmladapters/AwsStringToJavaDateTimeAdapter.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Date;
+
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+public class AwsStringToJavaDateTimeAdapter extends XmlAdapter<String, Date> {
+
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/sys/xmladapters/AwsStringToJavaDateTimeAdapter.java]: Can you move "yyyy-MM-dd HH:mm:ss" to a constant in CUKFSConstants .... there are some other date format strings in there.

**Comment by ajd299** [comment]
retest this please

**Comment by ajd299** [comment]
I've tried to address all of the issues, could you please take another look.
